library/game_state.py

```diff
@@ -131,15 +140,49 @@ def confirm_entry(self, term: blessed.Terminal) -> None:
 
         self.redraw_user_term(term)
 
-    def end_of_turn(self, term: blessed.Terminal) -> None:
+    def end_of_turn(self, term: blessed.Terminal, board: Board) -> None:
         """End of Turn"""
         self.current = 40
 
-        # update board
-        self.update_board = True
+        # Handle end of turn
+        working_space_location = convert_to_space_location(self.user_select_space)
+        subgrid_number = str(self.user_select_subgrid)
+        subgrid = board.collect_subgrid(subgrid_number)
+        if self.player_active == 1:
+            subgrid[working_space_location[0], working_space_location[1]] = "X"
+        elif self.player_active == 2:
+            subgrid[working_space_location[0], working_space_location[1]] = "O"
+
+        self.change_player()
+        if board.check_grid_victory(subgrid) == "X":
+            board.redraw_subgrid(term, subgrid, subgrid_number, term.green, "X")
+        elif board.check_grid_victory(subgrid) == "O":
+            board.redraw_subgrid(term, subgrid, subgrid_number, term.green, "O")
+        else:
+            board.redraw_subgrid(term, subgrid, subgrid_number, term.green, "None")
 
-        # circle back to top
-        self.wait_for_ready(term)
+        # handle logic for next grid
+
+        if board.check_subboard_victory((str(self.user_select_space))) not in (
+            "X",
+            "O",
+        ):
+            self.user_select_subgrid = self.user_select_space
+            self.user_select_space = 0
+            self.term_info[2] = (
+                f"Current: SubGrid {self.user_select_subgrid} "
+                f"| Space {self.user_select_space}"
+            )
+        else:
+            self.user_select_subgrid = 0
+        self.redraw_user_term(term)
+        self.update_board = False
+
+        # circle back to top or end the game
+        if board.check_board_victory() not in ("X", "O"):
+            self.wait_for_ready(term)
+        else:
+            self.game_over(term)
 
     def confirm_good_move(self, board: Board) -> bool:
         """Handle the entry of a bad move"""

```

Comment:
btw this should also check if the subgrid is already won

Reply:
This is the one thing I'm unclear of, this is only checking if the game is over. Checking if the subgrid has been won has happened earlier